### PR TITLE
ENH Update code to reflect changes in silverstripe/cms

### PR DESCRIPTION
**Pull request by GuySartorelli**
Replaces https://github.com/silverstripe/silverstripe-lumberjack/pull/175

Reflects changes in https://github.com/silverstripe/silverstripe-cms/pull/3017

## Issue
- https://github.com/silverstripe/silverstripe-cms/issues/2947

### CODE REVIEW

File: src/Forms/GridFieldSiteTreeEditButton.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function getColumnContent($gridField, $record, $columnName)
         // which can make the form readonly if no edit permissions are available.
 
         $data = ArrayData::create([
-            'Link' => $record->CMSEditLink(),
+            'Link' => $record->getCMSEditLink(),
```
Comment (GuySartorelli) [inline src/Forms/GridFieldSiteTreeEditButton.php:31]: Unrelated change - that method name changed in an earlier PR and we didn't catch this instance of it.

**Review by emteknetnz — APPROVED** [review]
